File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxA = 0;
        int left = 0;
        int right = height.length - 1;
        // for (int right = 0; right < height.length-1; right++) {
            while (right > left) {
                maxA = Math.max(maxA, (right - left) * Math.min(height[left], height[right]));
                if (height[left] < height[right]) {
                    left++;
                } else {
                    right--;
                }
            }
        
        return maxA;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var maxA = 0
        var left = 0
        var right = height.size - 1
        while (left < right) {
            maxA = Math.max(maxA, (right-left) * Math.min(height[left], height[right]))
            if (height[left] < height[right]) left++
            else right--
        }
        return maxA
    }
}